app/models/customer.rb

```diff
@@ -45,6 +47,20 @@ def full_name
     "#{first_name} #{last_name}".strip
   end
 
+  def self.find_or_new(email, enterprise_id)
+    Customer.new(email: email, enterprise_id: enterprise_id) unless email.present? && enterprise_id.present?
+
+    Customer.find_by(email: email, enterprise_id: enterprise_id) ||
+      Customer.new(email: email, enterprise_id: enterprise_id)
+  end
+
+  def set_created_manually_flag
+    self.created_manually = true
+    return unless persisted?
+
+    update_attribute(:created_manually, true)
```

Comment:
You are skipping validations here which is okay. But one may think that it only saves the one attribute to the database. In reality, Rails is assigning the attribute again and then saving all attributes to the database. In the way you use it, only this attribute is is changed but this may be confusing with future code changes. I would keep the logic the same but clarify what's happening:
```suggestion
    save(validate: false)
```

Alternatively, we could use `update_columns` which not only skips validation but also skips callbacks and writes only the given attributes to the database. Maybe this would be safer:
```suggestion
    update_columns(created_manually: true, updated_at: Time.zone.now)
```